code/entities/cooking/hotdog-roller/components/interactions/roller/Roller.cs

```diff
@@ -0,0 +1,156 @@
+﻿using Sandbox;
+using System.Collections.Generic;
+
+namespace Cinema;
+
+public partial class Roller : EntityComponent<HotdogRoller>
+{
+    private Dictionary<string, Hotdog> Hotdogs { get; set; } = new();
+
+    /// <summary>
+    /// Called when this component is activated
+    /// </summary>
+    protected override void OnActivate()
+    {
+        base.OnActivate();
+
+        TransitionStateTo(State.BothOff);
+    }
+
+    /// <summary>
+    /// Sets the roller's state
+    /// </summary>
+    public void SetPos(int pos)
+    {
+        State state = (State) pos;
+
+        TransitionStateTo(state);
+    }
+
+    /// <summary>
+    /// Adds hotdog by roller id
+    /// </summary>
+    public void AddHotdog(int rollerid)
```

Comment:
There is a bug with this function; If I add a hotdog while the machine is already on, it never starts rolling / steaming until i turn it off/on